lib/timeutils/tests/test_unixtime.c

```diff
@@ -44,6 +54,42 @@ Test(unixtime, unix_time_initialization)
   cr_assert(!unix_time_is_set(&ut));
 }
 
+Test(unixtime, unix_time_fix_timezone_adjusts_timestamp_as_if_was_parsed_assuming_the_incorrect_timezone)
+{
+  UnixTime ut = UNIX_TIME_INIT;
+  WallClockTime wct = WALL_CLOCK_TIME_INIT;
+
+  _wct_initialize(&wct, "Jan 19 2019 18:58:48");
+  wct.wct_gmtoff = 3600;
+  convert_wall_clock_time_to_unix_time(&wct, &ut);
+  cr_expect(ut.ut_gmtoff == 3600);
+
+  unix_time_fix_timezone(&ut, -5*3600);
```

Comment:
I don't understand the comment, sorry. This is using the conversion, as that's the key element of the timestamp guessing code. The hour of the day should not be changed, only the timezone.

Reply:
Simply put these test cases as is I think a conversion test, rather then a Unix time. There is a fine test set for them in  `test_conv.c`. So tests remain clean.
Still testing here `unix_time_fix_timezone` is ok, but only that, withouth any convertion:

Reply:
So you say, that instead of using a timestamp to initialize the Unix timestamp I should use an utc timestamp and that would make the test more readable?

I understand that the conversion is in a separate module, but surely a test depending on more modules doesn't introduce circular dependency. While in my opinion testing a bit more broadly we cover more code and the testcase effectively contains our main requirement.

The single requirement I want to test here is that the hour value remains and the timezone changes. How to do that without converting to/from hours.

If I avoid using wct, then all the test would contain is a change in the utc/timezone fields.

Of course that would be equivalent, but the main requirement would be completely missing from the test.

I feel that we are arguing a bit about how pure a unit test needs to be. I believe that they should encapsulate as much as needed, with the real code, especially if it makes the test requirements easier to code, and test run fast enough.

I remember the two camps of unit test practitioners, one camp always uses mocks so that we only use the subject under test the other incorporates, other, real classes as long as they are fast enough. I am a firm believer of the 2nd group. Here is a post that basically describes why, I am sure there are other explanations.

https://stackoverflow.com/questions/180413/mocks-or-real-classes

Reply:
Yes, if you initialise `UnixTime` directly, and call fix function, and do assert after; I think it can be as readable as asserting for hour, min, sec. As after calling fix function your requirement that the hour not changed, should already been met. The two case almost the same, but without `WallClockTime` it became simpler, thus more readable.

I am not worried here about circular dependencies, but rather if I change either `WallClockTime`, why should I change tests for `UnixTime` ? (in this case I might).

"I feel that we are arguing a bit about how pure a unit test needs to be." Mostly, and that's why I did not ask for change via change request.

While I do not see a relevance of mocking or not here; I agree on not mocking everything.